components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -1626,6 +1627,21 @@ private void addRoleInfo(String roleId, String roleName, List<Permission> permis
         }
     }
 
+    /**
+     * Check whether the corresponding application is a fragment application.
+     * @return
+     */
+    private boolean isFragmentApp() {
+
+        if (IdentityUtil.threadLocalProperties.get().get(IS_FRAGMENT_APP) != null) {
+            boolean isFragmentApp = Boolean.parseBoolean(IdentityUtil.threadLocalProperties.get().
+                    get(IS_FRAGMENT_APP).toString());
+            IdentityUtil.threadLocalProperties.get().remove(IS_FRAGMENT_APP);
```

Comment:
This is not a proper place to clear the threadlocal.
Can we guarantee that always this point is reached after the thread local is set?
If any exception is thrown from one of above logic before this point (existing/ future codes), the threadlocal will not get cleared.

Reply:
Since the pre listener which invokes the thread local setting method is invoked before from addRole method,
clear this threadlocal at the finally block of the addRole method

Also for safe side, in the pre listner side as well clear the threadlocal before setting it

Reply:
Updated